frontend/src/app/versions.tsx

```diff
@@ -419,29 +432,36 @@ class Versions extends Component<IProps, any> {
     }
 
     private changePublishState = (buttonText) => {
-        const formData = new FormData();
-        if (buttonText === "Publish") {
-            formData.append(":operation", "pant:release");
-            // console.log('Published file path:', this.props.modulePath)
-            this.draft[0].version = "";
+        // Validate productValue before Publishs
+        if (this.props.productInfo !== undefined && this.props.productInfo.trim() === "" && buttonText === "Publish") {
```

Comment:
If `productInfo` IS undefined, wouldn't we want to show the dialog in that scenario as well?

Reply:
Actually, now that I'm thinking about it, I would really like this to be a server-side check. Although 99% of the time the users will be publishing from our web UI - we can't guarantee that that will always be the case. I'd like to see the server providing a computed property with the metadata - something like `canPublish` - that the UI keys off of, rather than making the decision at the UI layer like you have here. Can you implement that?

Reply:
Ben, 

I agree that we should have a server side validation. Let's log a jira to capture that effort so that we don't expand the scope of this jira.

Reply:
I agree that the server-side check is extra effort, but I don't like the idea of committing this UI-side check with the intention of fixing it up properly later (but maybe never). I checked the story in Jira and it doesn't specify one way or the other, so I'd really prefer that we implement this the right way the first time. If we need to repoint the story as a result of the increased effort, then we can do so.

Reply:
CCS 2962 is updated to reflect that UI work. 
I created a new jira for the backend effort: https://projects.engineering.redhat.com/browse/CCS-3237

Reply:
Well, I still think we could have resized the original story, but that's a small detail. Just to be sure, is your intention to work that new story before moving forward with this PR?

Reply:
My goal is to get this one moved forward then address other stories.

Reply:
Ben, per our offline conversation, I pushed up the validation for ReleaseDraftVersion servlet